addons/stock/models/stock_rule.py

```diff
@@ -584,8 +567,25 @@ def _procure_orderpoint_confirm(self, use_new_cursor=False, company_id=None):
                 else:
                     raise
 
+            # Log an activity on product template for failed orderpoints.
             for orderpoint, error_msg in orderpoints_exceptions:
-                self.env['stock.rule']._log_next_activity(orderpoint.product_id, error_msg)
+                existing_activity = self.env['mail.activity'].search([
+                    ('res_id', '=', orderpoint.product_id.product_tmpl_id.id),
+                    ('res_model_id', '=', self.env.ref('product.model_product_template').id),
+                    ('note', '=', error_msg)])
+                if not existing_activity:
+                    # If the user deleted warning activity type.
+                    try:
+                        activity_type_id = self.env.ref('mail.mail_activity_data_warning').id
+                    except:
```

Comment:
except what?

Reply:
sorry you juste moved the code.. you could replace the try except by activity_type_id = self.env.ref('mail.mail_activity_data_warning', raise_if_no_found=False)